src/OrchardCore.Modules/OrchardCore.ContentFields/Controllers/ContentPickerAdminController.cs

```diff
@@ -75,7 +86,23 @@ public async Task<IActionResult> SearchContentItems(string part, string field, s
                 PartFieldDefinition = partFieldDefinition
             });
 
-            return new ObjectResult(results.Select(r => new VueMultiselectItemViewModel() { Id = r.ContentItemId, DisplayText = r.DisplayText, HasPublished = r.HasPublished }));
+            var contentItems = await _contentManager
+                .GetAsync(results.Select(r => r.ContentItemId));
```

Comment:
Instead of pulling the content items here all over again, modify the result-provides (for example, `DefaultContentPickerResultProvider`). Add `IsViewable` property to `ContentPickerResult` class and preset the permission there.

Reply:
I would need to change the other implementations of `IContentPickerResultProvider` then, too. Is that fine? I didn't want to introduce so many changes for a field which is used in the `ContentPickerField.Edit` only. Also `IContentManager` shouldn't need to query those items again, since they should be cached? But if making the change for all `IContentPickerResultProvider` is fine, I can do it that way

Reply:
We already do other checks against the contentItem like `HasPublished` in the provider